pkg/controller/endpointslice/utils.go

```diff
@@ -267,6 +270,54 @@ func serviceControllerKey(endpointSlice *discovery.EndpointSlice) (string, error
 	return fmt.Sprintf("%s/%s", endpointSlice.Namespace, serviceName), nil
 }
 
+// setEndpointSliceLabels returns a map with the new endpoint slices labels and true if there was an update.
+// Slices labels must be equivalent to the Service labels except for the reserved IsHeadlessService, LabelServiceName and LabelManagedBy labels
+// Changes to IsHeadlessService, LabelServiceName and LabelManagedBy labels on the Service do not result in updates to EndpointSlice labels.
+func setEndpointSliceLabels(epSlice *discovery.EndpointSlice, service *corev1.Service) (map[string]string, bool) {
+	updated := false
+	epLabels := make(map[string]string)
+
+	// check if the endpoint slice and the service have the same labels
+	// clone current slice labels except the reserved labels
+	for key, value := range epSlice.Labels {
+		if key == discovery.LabelServiceName ||
+			key == discovery.LabelManagedBy ||
+			key == v1.IsHeadlessService {
+			continue
+		}
+		// copy endpoint slice labels
+		epLabels[key] = value
+	}
+
+	// if the labels are not identical update the slice with the corresponding service labels
+	if !apiequality.Semantic.DeepEqual(epLabels, service.Labels) {
+		updated = true
+		epLabels = make(map[string]string)
+		// add service labels
+		for key, value := range service.Labels {
+			// don't copy reserved labels
+			if key == discovery.LabelServiceName || key == discovery.LabelManagedBy {
+				klog.Warningf("Service %s/%s using reserved endpoint slices label, skipping label %s: %s", service.Namespace, service.Name, key, value)
+				continue
+			}
```

Comment:
Maybe we can add a helper function that can be used here and above to indicate if a label is reserved? Something like `isReservedLabelKey(key)`

Reply:
yeah, that will make it cleaner, and can be reused